src/Signature.ts

```diff
@@ -85,18 +85,20 @@ export class Signature {
         return this.signature.type;
     }
 
-    /** Verify a signature with a message digest and public key */
-    public verify(digest: BNInput, publicKey: PublicKey, encoding?: string): boolean {
+    /** Verify a signature with a message or hashed message digest and public key */
+    public verify(data: BNInput, publicKey: PublicKey, shouldHash: boolean = true, encoding: string = 'utf8'): boolean {
+        if (shouldHash) data = this.ec.hash().update(data, encoding).digest();
         const ellipticSignature = this.toElliptic();
         const ellipticPublicKey = publicKey.toElliptic();
-        return this.ec.verify(digest, ellipticSignature, ellipticPublicKey, encoding);
+        return this.ec.verify(data, ellipticSignature, ellipticPublicKey, encoding);
     }
 
-    /** Recover a public key from a message digest and signature */
-    public recoverPublicKey(digest: BNInput, encoding?: string): PublicKey {
+    /** Recover a public key from a message or hashed message digest and signature */
+    public recover(data: BNInput, shouldHash: boolean = true, encoding: string = 'utf8'): PublicKey {
```

Comment:
`shouldHash` default should be false for backwards compatibility. Looks like having an `encoding` non-default needs to happen too unless `elliptic` uses `utf8` by default.

Reply:
`eosjs-ecc` `recover` hashes the data by default, hence `shouldhash` is set to true by default.

Reply:
This comment was only related to backwards compatibility if we released this as `21.x.0`